### PR TITLE
provider/openstack: add created time to load balancer definition

### PR DESCRIPTION
**Pull request by drmaas**

### CODE REVIEW

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackNetworkingV2Provider.groovy [hunk 1]
```diff
@@ -36,15 +37,10 @@ import org.openstack4j.model.network.ext.Member
...
 
-import java.util.regex.Matcher
-import java.util.regex.Pattern
-
-class OpenstackNetworkingV2Provider implements OpenstackNetworkingProvider, OpenstackRequestHandler, OpenstackIdentityAware {
+class OpenstackNetworkingV2Provider implements OpenstackNetworkingProvider, OpenstackRequestHandler, OpenstackIdentityAware, LoadBalancerResolver {
 
   final int minPort = 1
   final int maxPort = (1 << 16) - 1
-  final String lbDescriptionRegex = ".*internal_port=([0-9]+).*"
```
Comment (varikin) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackNetworkingV2Provider.groovy:46]: Did this get moved somewhere or do you not need it anymore?

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/domain/LoadBalancerPool.groovy [hunk 2]
```diff
@@ -41,7 +42,22 @@ class LoadBalancerPool implements LoadBalancerResolver {
...
-    this.description = "internal_port=${internalPort}"
+    String portString = generateInternalPort(port)
+    if (this.description) {
+      this.description += ",$portString"
+    } else {
+      this.description = portString
+    }
+  }
+
+  void setCreatedTime(long time) {
```
Comment (varikin) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/domain/LoadBalancerPool.groovy]: If we need to keep adding things to the description, I would recommend a method for appending to the description. With just two methods, I am not worried yet.

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/domain/LoadBalancerResolver.groovy [hunk 3]
```diff
@@ -19,31 +19,99 @@ package com.netflix.spinnaker.clouddriver.openstack.domain
...
+  /**
+   * Parse the internal port from a load balancer description in the following format.
+   * <br><br>
+   * {@code
+   *  ...,internal_port=8100,...
+   * }
+   * @param description
+   * @return the port value
+   */
   int getInternalPort(final String description) {
```
Comment (varikin) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/domain/LoadBalancerResolver.groovy]: Ah! Here is where that ended up. Nice place for it.

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/domain/LoadBalancerResolver.groovy [hunk 4]
```diff
@@ -19,31 +19,99 @@ package com.netflix.spinnaker.clouddriver.openstack.domain
...
+  /**
+   * Parse the created time from a load balancer description in the following format.
+   * <br><br>
+   * {@code
+   *  ...,created_time=12345678,...
+   * }
+   * @param description
+   * @return the port value
+   */
+  long getCreatedTime(final String description) {
```
Comment (varikin) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/domain/LoadBalancerResolver.groovy:67]: I wonder if the `setCreatedTime` and `setInternalPort` should go in this class as well

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/ops/loadbalancer/UpsertOpenstackLoadBalancerAtomicOperation.groovy [hunk 5]
```diff
@@ -73,6 +73,7 @@ class UpsertOpenstackLoadBalancerAtomicOperation implements AtomicOperation<Map>
       if (newLoadBalancerPool.id) {
         resultPool = updateLoadBalancer(region, newLoadBalancerPool, virtualIP, poolHealthMonitor)
       } else {
+        newLoadBalancerPool.createdTime = System.currentTimeMillis()
```
Comment (derekolk) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/ops/loadbalancer/UpsertOpenstackLoadBalancerAtomicOperation.groovy]: Why not do this in the createLoadBalancer method?

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackNetworkingV2Provider.groovy (hunk 1 above)
Comment (drmaas) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackNetworkingV2Provider.groovy:46]: correct

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/domain/LoadBalancerPool.groovy (hunk 2 above)
Comment (derekolk) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/domain/LoadBalancerPool.groovy]: Agreed ... This should be DRY'd up.

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/domain/LoadBalancerResolver.groovy (hunk 4 above)
Comment (drmaas) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/domain/LoadBalancerResolver.groovy:67]: have methods to generate them in the resolver. The setters get called when building objects.

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/ops/loadbalancer/UpsertOpenstackLoadBalancerAtomicOperation.groovy (hunk 5 above)
Comment (drmaas) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/ops/loadbalancer/UpsertOpenstackLoadBalancerAtomicOperation.groovy]: could.

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/domain/LoadBalancerResolver.groovy [hunk 6]
```diff
@@ -19,31 +19,99 @@ package com.netflix.spinnaker.clouddriver.openstack.domain
...
+   * }
+   * @param description
+   * @return the port value
+   */
   int getInternalPort(final String description) {
-    int result = -1
-    if (description) {
-      Matcher matcher = descriptionPattern.matcher(description)
+    String s = match(description, portPattern)
+    if (s) {
```
Comment (derekolk) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/domain/LoadBalancerResolver.groovy]: Isn't there groovy syntax sugar for this?  s ? s.toInteger() : -1

Comment (drmaas) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/domain/LoadBalancerResolver.groovy]: ok

**Comment by derekolk** [comment]
LGTM :+1

**Pull request closed without merging**